frontend/components/redeem/CampaignCharitySelection.tsx

```diff
@@ -0,0 +1,61 @@
+import { Grid, Radio, Stack, Typography } from '@mui/material';
+import { containerSx } from '../../styles/redeem/indexStyles';
+import { CouponRedeemData } from '../../types/coupons';
+import { Nullable } from '../../types/utils';
+import { theme } from '../../utils/theme';
+import CampaignDescription from '../campaigns/CampaignDescription';
+import CampaignCharityCard from '../charities/CampaignCharityCard';
+import CampaignCharityList from '../charities/CampaignCharityList';
+import Button from '../generic/Button';
+
+type Props = {
+  coupon: CouponRedeemData;
+  campaignCharityId: Nullable<number>;
+  setCampaignCharityId: (campaignCharityId: number) => void;
+  goToNextPage: () => void;
+};
+
+const CampaignCharitySelection = ({ coupon, campaignCharityId, setCampaignCharityId, goToNextPage }: Props) => {
+  return (
+    <Grid container sx={containerSx} component="main" justifyContent="center" padding={2}>
+      <Grid item md={12} lg={4}>
+        <Stack spacing={theme.spacing(2)} paddingBottom={4}>
```

Comment:
I realised the proper way is to use useTheme, and pass theme into SxProps objects through their function if needed. This is so we can swap themes easily like dark vs light theme through the `ThemeProvider` at `app.tsx`. Not changing the styles files yet, but for .tsx files let's use useTheme.

```suggestion
const CampaignCharitySelection = ({ coupon, campaignCharityId, setCampaignCharityId, goToNextPage }: Props) => {
  const theme = useTheme();
  return (
    <Grid container sx={containerSx} component="main" justifyContent="center" padding={2}>
      <Grid item md={12} lg={4}>
        <Stack spacing={theme.spacing(2)} paddingBottom={4}>
```